### PR TITLE
trim and align top-level build scripts

### PR DESCRIPTION
**Pull request by adamralph**
**What issue does this PR address?**

I have a couple of crazy ideas regarding the top-level and solution-specific build scripts, but I wanted to first clean a few things up and confirm my assumptions about how things are working.

**Does this PR introduce a breaking change?**

No.

**Please check if the PR fulfills these requirements**
- [x] The commit follows our [guidelines](https://github.com/IdentityServer/IdentityServer4/blob/master/.github/CONTRIBUTING.md)
- [x] ~Unit Tests for the changes have been added (for bug fixes / features)~

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -1,80 +1,25 @@
+$ErrorActionPreference = "Stop";
```
Comment (adamralph) [inline build.ps1]: This should replace all the explicit error checks.

File: build.ps1 [hunk 2]
```diff
@@ -1,80 +1,25 @@
...
 dotnet tool restore
 
-$cd = Get-Location
-
-# storage
-""
-"###########################################"
-"######### IdentityServer4.Storage #########"
-"###########################################"
-""
```
Comment (adamralph) [inline build.ps1]: Now that the solution is identified in the build output, these headers should not be needed.

image

File: build.sh [hunk 3]
```diff
@@ -1,56 +1,26 @@
-mkdir nuget
+#!/usr/bin/env bash
+set -euo pipefail
```
Comment (adamralph) [inline build.sh]: This should replace the explicit error checks.

File: build.sh [hunk 4]
```diff
@@ -1,56 +1,26 @@
...
 
-cd ./src/Storage
-./build.sh
+mkdir nuget -p
```
Comment (adamralph) [inline build.sh]: `-p` ensure the command does not fail if the directory already exists.

File: build.sh [hunk 5]
```diff
@@ -1,56 +1,26 @@
...
 
-if [ $? -ne 0 ]; then
-    echo "An error occured."
-    exit 1
-fi
+dotnet tool restore
```
Comment (adamralph) [inline build.sh]: For consistency with `build.ps1`.

File: build.sh [hunk 6]
```diff
@@ -1,56 +1,26 @@
...
-cd ./src/EntityFramework.Storage
-./build.sh
-
-if [ $? -ne 0 ]; then
-    echo "An error occured."
-    exit 1
-fi
-
-cd ../..
-cp ./src/EntityFramework.Storage/artifacts/* ./nuget
```
Comment (adamralph) [inline build.sh]: The individual build scripts are already doing this work.

**Comment by leastprivilege** [comment]
thanks!

https://twitter.com/leastprivilege/status/1183624314576408577

;)

**Pull request merged**